test/refactoring/AddCodableImplementation/basic.swift

```diff
@@ -0,0 +1,22 @@
+// RUN: rm -rf %t.result && mkdir -p %t.result
+
+struct User: Codable {
+  let firstName: String
+  let lastName: String?
+}
+// RUN: %refactor -add-explicit-codable-implementation -source-filename %s -pos=3:8 > %t.result/codable.swift
```

Comment:
Alex added a nice `refactor-check-compiles` that we use for the convert async refactorings. I also prefer `CHECK` to output + diff, but maybe I'm the only one 😆

Reply:
Outputting each run to a separate file makes it very easy to spot bugs (and e.g. to copy the code). Having that all inline in comments would make it harder to understand, I think. 🙂

Reply:
It's not too bad for these, but for many refactorings *most* of the output doesn't actually matter. I guess this one has a lot compared to the file as a whole though. And personally I find copy pasting often misses bugs 🤷.

Anyway, I'm fine with the separate file, but I would like to check that it compiles.

Reply:
> but I would like to check that it compiles.

As we discussed offline, `refactor-check-compiles` doesn't currently work when outputting to a file. That might be a trivial fix to the test infrastructure, but not something I want to attempt in this PR.